bindings/rust/s2n-tls/src/connection.rs

```diff
@@ -969,6 +969,47 @@ impl Connection {
         }
         Ok(secret)
     }
+
+    /// Retrieves the size of the serialized connection
+    pub fn serialized_connection_length(&self) -> Result<u32, Error> {
+        unsafe {
+            let mut length = 0;
+            s2n_connection_serialization_length(self.connection.as_ptr(), &mut length)
+                .into_result()?;
+            Ok(length)
+        }
+    }
+
+    /// Serializes the TLS connection into the provided buffer
+    pub fn serialize_connection(&self, output: &mut [u8]) -> Result<(), Error> {
+        unsafe {
+            s2n_connection_serialize(
+                self.connection.as_ptr(),
+                output.as_mut_ptr(),
+                output.len().try_into().map_err(|_| Error::INVALID_INPUT)?,
+            )
+            .into_result()?;
+            Ok(())
+        }
+    }
+
+    /// Deserializes the input buffer into a new TLS connection that can send/recv
+    /// data from the original peer.
+    pub fn deserialize_connection(&mut self, input: &[u8]) -> Result<(), Error> {
```

Comment:
Should this allocate the connection for the caller and return that connection?

Reply:
I can do this if the function takes in a Connection::Mode parameter, otherwise I won't know if I should create a server or a client.